### PR TITLE
Hygiene fixes to Logging Framework

### PR DESCRIPTION
**Pull request by aggarw13**
*Issue #, if available:*

*Description of changes:*
* ~~Update the common logging interface to accept the library name string as a parameter for better config.h file readability.~~ (Unchanged based on review feedback)
* Some hygiene fixes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/config.h [hunk 1]
```diff
@@ -29,16 +29,16 @@ typedef int MQTTNetworkContext_t;
...
 
 /* Define the IotLog logging interface to enabling logging.
  * This demo maps the macro to the reference POSIX implementation for logging. */
-#define IotLog( messageLevel, pFormat, ... ) \
-    IotLog_Generic( messageLevel,            \
-                    "[%s:%d] [%s] "pFormat,  \
-                    __FILE__,                \
-                    __LINE__,                \
-                    LIBRARY_LOG_NAME,        \
+#define IotLog( messageLevel, pLibraryName, pFormat, ... ) \
```
Comment (aggarg) [inline demos/config.h:36]: I do not think that we should add ```pLibraryName``` as a parameter to the ```IotLog``` macro. A customer can very well choose not to use ```pLibraryName```  in their logs by defining ```IotLog``` like:

```c
#define IotLog( messageLevel, pFormat, ... ) \
    IotLog_Generic( messageLevel,            \
                    "[%s:%d] "pFormat,  \
                    __FILE__,                \
                    __LINE__,                \
                    __VA_ARGS__ )
```
The customer will still be forced to define ```LIBRARY_LOG_NAME``` like:

```
#define LIBRARY_LOG_NAME "Customer App"
#include "iot_logging_setup.h"
```

On the other hand, if we kept it like before and removed the ```#ifndef LIBRARY_LOG_NAME``` check from iot_logging_setup.h, the customer who chooses not to include library names in their logs, will not be forced to define ```LIBRARY_LOG_NAME```.

Comment (cobusve) [reply demos/config.h:36]: +1 to that. I think also this interface change will break all code that uses this function directly.  I found a bunch of places like this https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/2607cd8d609ef4fc61ff67a42eb1d6c2f7d420f2/libraries/standard/mqtt/src/iot_mqtt_serialize.c#L120

Comment (aggarw13) [reply demos/config.h:36]: The motivation behind adding the `pLibraryName` parameter to the interface was to remove exposing the `LIBRARY_LOG_NAME` macro in the `config.h` which may not carry any meaning to the customer, who looks at the config file without knowledge of the internal details of the logging macro framework. 
@cobusve, regarding the example shared on the breaking change, this PR is destined for the `development` branch where the new MQTT code, being re-written from scratch (along with the logging calls), is not expected to call the old `IotLog` interface that exists in the `v4_beta` branch.

File: libraries/standard/utilities/include/iot_logging_setup.h [hunk 2]
```diff
@@ -114,7 +114,7 @@
  */
 
 /**
- * @def IotLog( messageLevel, pLogConfig, ... )
+ * @def IotLog( messageLevel, pLibraryName, pFormat, ... )
```
Comment (sarenameas) [inline libraries/standard/utilities/include/iot_logging_setup.h:117]: Given the outcome, this change needs to be reverted?

Comment (aggarw13) [reply libraries/standard/utilities/include/iot_logging_setup.h]: Yes, reverted

File: demos/include/config.h [hunk 3]
```diff
@@ -29,10 +29,12 @@ typedef int MQTTNetworkContext_t;
 #define MQTT_MAX_QUEUED_PUBLISH_MESSAGES    10
 
 /* Include file for POSIX reference implementation. */
-#include "port/posix/iot_logging.h"
+#include "platform/include/iot_logging.h"
 
 /* Define the IotLog logging interface to enabling logging.
- * This demo maps the macro to the reference POSIX implementation for logging. */
+ * This demo maps the macro to the reference POSIX implementation for logging.
+ * Note: @ref LIBRARY_LOG_LEVEL adds the name of the library, that produces the
```
Comment (aggarg) [inline demos/include/config.h:36]: LIBRARY_LOG_LEVEL  --> LIBRARY_LOG_NAME

Comment (aggarw13) [reply demos/include/config.h]: Fixed

**Pull request merged**